### PR TITLE
Improve machine data handling when exporting uncommitted models.

### PR DESCRIPTION
**Pull request by frankban**
Fixes https://github.com/juju/juju-gui/issues/2801
Also fixes constraints and series not being exported before committing machines manually created from the machine view.

### CODE REVIEW

**Comment by frankban** [comment]
To QA, follow the instructions in the linked bug.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7274/

File: jujugui/static/gui/src/app/models/models.js [hunk 1]
```diff
@@ -2616,24 +2533,24 @@ YUI.add('juju-models', function(Y) {
...
-                  }
-                  return parts.join(':');
-                } else {
-                  return machineIdMap[machineId] + '';
-                }
-              });
-        }
+        serviceList[serviceName].to = machinePlacement[serviceName].map(
+          function(machineId) {
+            var parts = machineId.split(':');
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/models/models.js]: `let`

File: jujugui/static/gui/src/app/models/models.js [hunk 2]
```diff
@@ -2616,24 +2533,24 @@ YUI.add('juju-models', function(Y) {
...
+            if (parts.length === 2) {
+              // It's a container
+              var partInt = parseInt(parts[1], 10);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/models/models.js]: `let`

File: jujugui/static/gui/src/app/models/models.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/app/models/models.js]: Done.

File: jujugui/static/gui/src/app/models/models.js (hunk 2 above)
Comment (frankban) [inline jujugui/static/gui/src/app/models/models.js]: Done.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7289/

File: jujugui/static/gui/src/test/test_model.js [hunk 3]
```diff
@@ -2468,8 +2399,8 @@ describe('test_model.js', function() {
           }
         }
       }]);
-      var result = db.exportDeployer(true);
-      assert.deepEqual(result.services.mysql.to, ['0']);
+      var result = db.exportBundle();
```
Comment (Lukewh) [inline jujugui/static/gui/src/test/test_model.js]: const

Comment (frankban) [inline jujugui/static/gui/src/test/test_model.js]: Done.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7294/

**Comment by Lukewh** [comment]
QAOK @frankban

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**